assets/src/admin/site-scan-notice/notice.js

```diff
@@ -0,0 +1,124 @@
+/**
+ * WordPress dependencies
+ */
+import { useContext, useEffect, useMemo } from '@wordpress/element';
+import { __, sprintf } from '@wordpress/i18n';
+
+/**
+ * External dependencies
+ */
+import {
+	AMP_COMPATIBLE_PLUGINS_URL,
+	SETTINGS_LINK,
+} from 'amp-site-scan-notice'; // From WP inline script.
+
+/**
+ * Internal dependencies
+ */
+import { SiteScan } from '../../components/site-scan-context-provider';
+import { User } from '../../components/user-context-provider';
+import {
+	ADMIN_NOTICE_TYPE_ERROR,
+	ADMIN_NOTICE_TYPE_INFO,
+	ADMIN_NOTICE_TYPE_SUCCESS,
+	ADMIN_NOTICE_TYPE_WARNING,
+	AdminNotice,
+} from '../../components/admin-notice';
+import { Loading } from '../../components/loading';
+import { isExternalUrl } from '../../common/helpers/is-external-url';
+import { PluginsWithAmpIncompatibility } from './plugins-with-amp-incompatibility';
+
+// Define Plugin Suppression link.
+const PLUGIN_SUPPRESSION_LINK = new URL( SETTINGS_LINK );
+PLUGIN_SUPPRESSION_LINK.hash = 'plugin-suppression';
+
+export function SiteScanNotice() {
+	const {
+		cancelSiteScan,
+		fetchScannableUrls,
+		isCancelled,
+		isCompleted,
+		isFailed,
+		isInitializing,
+		isReady,
+		pluginsWithAmpIncompatibility,
+		startSiteScan,
+	} = useContext( SiteScan );
+	const { developerToolsOption } = useContext( User );
+	const userIsTechnical = useMemo( () => developerToolsOption === true, [ developerToolsOption ] );
+
+	// Cancel scan on component unmount.
+	useEffect( () => cancelSiteScan, [ cancelSiteScan ] );
+
+	// Fetch scannable URLs on mount. Start site scan right after the component is mounted and the scanner is ready.
+	useEffect( () => {
+		if ( isInitializing ) {
+			fetchScannableUrls();
+		} else if ( isReady ) {
+			startSiteScan();
+		}
+	}, [ fetchScannableUrls, isInitializing, isReady, startSiteScan ] );
+
+	const commonNoticeProps = {
+		className: 'amp-site-scan-notice',
+		isDismissible: true,
+		onDismiss: cancelSiteScan,
+	};
+
+	if ( isFailed || isCancelled ) {
+		return (
+			<AdminNotice type={ ADMIN_NOTICE_TYPE_ERROR } { ...commonNoticeProps }>
+				<p>
+					{ __( 'AMP plugin could not check your site for compatibility issues.', 'amp' ) }
+				</p>
+			</AdminNotice>
+		);
+	}
+
+	if ( isCompleted && pluginsWithAmpIncompatibility.length === 0 ) {
+		return (
+			<AdminNotice type={ ADMIN_NOTICE_TYPE_SUCCESS } { ...commonNoticeProps }>
+				<p>
+					{ __( 'AMP plugin found no validation errors.', 'amp' ) }
+				</p>
+			</AdminNotice>
+		);
+	}
+
+	if ( isCompleted && pluginsWithAmpIncompatibility.length > 0 ) {
+		return (
+			<AdminNotice type={ ADMIN_NOTICE_TYPE_WARNING } { ...commonNoticeProps }>
+				<p
+					dangerouslySetInnerHTML={ {
+						__html: sprintf(
+							/* translators: the placeholder is a link to the Plugin Suppression Settings panel. */
+							__( 'AMP Plugin found validation errors. <a href="%s">Review Plugin Suppression Settings</a>', 'amp' ),
+							PLUGIN_SUPPRESSION_LINK,
+						),
+					} }
+				/>
```

Comment:
See my [latest comment](https://github.com/ampproject/amp-wp/pull/6685#issuecomment-983612091).

Reply:
> (Otherwise, shouldn't `createInterpolateElement` be used? Or is that not safe since we may not be able to polyfill on this screen?)

From what I can tell so far, the `createInterpolateElement` is useful when interpolating anything that is non-trivial. E.g. `<a href="#">...</a>` or `<b>...</b>` are simple elements and for them `dangerouslySetInnerHTML` + `sprintf` is sufficient. If we wanted to interpolate a React component or an HTML element with some conditional/complex attributes (like an external link), then `createInterpolateElement` would be the way to go for sure.

Reply:
Isn't `createInterpolateElement` also preferable for security reasons since it avoids the use of `RawHTML` or `dangerouslySetInnerHTML`? This is what I understood from https://make.wordpress.org/core/2020/07/17/introducing-createinterpolateelement/

Reply:
That's a very good point that I missed. Security-wise, it's better now to use `createInterpolateElement` instead of `dangerouslySetInnerHTML`. We had to resort to `dangerouslySetInnerHTML` before since there was no better option.

I think I'll issue a PR where the occurrences of `dangerouslySetInnerHTML` will be replaced with `createInterpolateElement` in the entire codebase.

Reply:
I've created an issue for that: #6759.